src/planning.rs

```diff
@@ -100,12 +100,10 @@ impl<'a> BuildPlanner<'a> {
 
     let mut crate_contexts = Vec::new();
 
-    let source_id = match self.registry.clone() {
-      Some(v) => v,
-      None => try!(SourceId::crates_io(&self.cargo_config)),
-    };
-
-    let package_ids = try!(find_all_package_ids(source_id, &resolve));
+    let mut package_ids: Vec<PackageId> = Vec::new();
```

Comment:
If `Remote` genmode is currently unsupported, perhaps we should fail harder. Something like

```rust
// N.B. Renamed for clarity
let upstream_registry_source_id = try!(SourceId::crates_io(&self.cargo_config));

// Verify that Remote genmode only has crates_io deps (others currently unsupported)
if self.settings.genmode == GenMode::Remote { 
  for dep in root_direct_deps.iter() {
    if dep != root_package_id && dep.source_id() != upstream_registry_source_id {
      panic!("Remote GenMode doesn't currently support crate sources other than crates_io (e.g. `git` is unsupported)");
    }
  }
}
```

Reply:
Agreed, and that's straight-forward to change. I want to test that I didn't accidentally introduce support for it in remote GenMode (and will help me figure out what is needed there), but following confirmation that it's broken, will add a failback.